certbot/certbot/_internal/main.py

```diff
@@ -1090,7 +1089,12 @@ def revoke(config, unused_plugins):
     config.installer = config.authenticator = None
 
     if config.cert_path is None and config.certname:
-        config.cert_path = storage.cert_path_for_cert_name(config, config.certname)
+        # When revoking via --cert-name, take the cert path and server from renewalparams
+        lineage = storage.RenewableCert(
+            storage.renewal_file_for_certname(config, config.certname), config)
+        config.cert_path = lineage.cert_path
+        if not cli.set_by_cli("server"): # --server should override the lineage server
```

Comment:
It may be overkill, but what do you think about also checking for the case where `lineage.server` is `None`? From 0.5.0 to 0.26.0 we didn't save `server` values by default. See https://github.com/certbot/certbot/issues/7978#issuecomment-625442870.